libyul/CMakeLists.txt

```diff
@@ -127,6 +127,8 @@ add_library(yul
 	optimiser/FullInliner.h
 	optimiser/FunctionCallFinder.cpp
 	optimiser/FunctionCallFinder.h
+	optimiser/FunctionDefinitionCollector.cpp
```

Comment:
At some point we probably should split this directory into files / classes that only perform analysis or are only utility classe and actual optimizer steps.